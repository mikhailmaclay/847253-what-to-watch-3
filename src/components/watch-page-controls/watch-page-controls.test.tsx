// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// Components
import WatchPageControls from './watch-page-controls';
//
import emptyArrowFunction from '../../utils/components/empty-arrow-function';

describe(`<WatchPageControls/>`, () => {
  const props = {
    name: `Test`,
    isFullscreen: false,
    isPlaying: true,
    isMetadataLoaded: true,
    isWaiting: false,
    isPaused: false,
    isEnded: false,
    isWatchMode: false,
    currentTime: 1,
    duration: 100,
    onFullscreenButtonClick: emptyArrowFunction,
    onPlayButtonClick: emptyArrowFunction,
    onProgressBarClick: emptyArrowFunction,
    onProgressBarMouseOver: emptyArrowFunction,
    onProgressBarButtonMouseDown: emptyArrowFunction,
    onClose: emptyArrowFunction,
  };

  it(`Should render correctly`, () => {
    const result = renderer.create(
        <WatchPageControls {...props}/>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
