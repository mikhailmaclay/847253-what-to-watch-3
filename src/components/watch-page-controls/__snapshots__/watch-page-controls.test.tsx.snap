// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchPageControls/> Should render correctly 1`] = `
Array [
  <button
    className="player__exit"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    type="button"
  >
    Exit
  </button>,
  <div
    className="player__controls"
    style={
      Object {
        "backgroundImage": "linear-gradient(transparent, black)",
        "zIndex": "999",
      }
    }
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={100}
          onClick={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          value={1}
        />
        <div
          className="player__toggler"
          onMouseDown={[Function]}
          style={
            Object {
              "left": "1%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:01:39
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        title="Pause"
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Test
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        title="Fullscreen"
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>,
]
`;
